### PR TITLE
Insert `static` when fully qualifying in a `using` declaration

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/49986

### CODE REVIEW

File: src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs [hunk 1]
```diff
@@ -19,14 +17,21 @@ internal abstract partial class AbstractFullyQualifyCodeFixProvider : CodeFixPro
...
+            /// the <see cref="WithSymbol(INamespaceOrTypeSymbol)"/> method and doesn't affect comparisons.
+            /// </summary>
+            public readonly bool OriginalSymbolIsType;
+
+            public SymbolResult(INamespaceOrTypeSymbol symbol, int weight, bool originalSymbolIsType)
             {
                 Symbol = symbol;
                 Weight = weight;
                 NameParts = INamespaceOrTypeSymbolExtensions.GetNameParts(symbol);
+                OriginalSymbolIsType = originalSymbolIsType;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: looks odd seeing this value passed in.  can't it be inerred from 'symbol' instead?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: looking through the PR, i think the answer is: yes.

Comment (davidwengier) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: Sadly no, Symbol is reused to represent the containing symbol, [here](https://github.com/dotnet/roslyn/blob/master/src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.cs#L315) so by the time we need to use it `Symbol.IsType` will always be `false`.

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: :'(   ok :)

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: alternatively, could we jus tpass that symbol along?

Comment (davidwengier) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: Yeah, can pass around both symbols if you think its better.

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/FullyQualify/AbstractFullyQualifyCodeFixProvider.SymbolResult.cs:31]: I think i would personally prefer that.

**Pull request merged**